otcextensions/sdk/dis/v2/_proxy.py

```diff
@@ -10,8 +10,273 @@
+            delete a nonexistent app.
+
+        :returns: ``None``
+        """
+        return self._delete(_app.App, app, ignore_missing=ignore_missing)
+
+    # ======== Stream ========
+    def add_checkpoint(self, **attrs):
+        """Add a Checkpoint from attributes
+
+        :param dict attrs: Keyword arguments which will be used to submit
+            a :class:`~otcextensions.sdk.dis.v2.checkpoint.Checkpoint`,
+            comprised of the properties on the Checkpoint class.
+        :returns: The results of stream creation
+        :rtype: :class:`~otcextensions.sdk.dis.v2.checkpoint.Checkpoint`
+        """
+        attrs.update(checkpoint_type='LAST_READ')
+        return self._create(_checkpoint.Checkpoint, **attrs)
+
+    def get_checkpoint(self, stream_name, app_name,
+                       partition_id, checkpoint_type='LAST_READ'):
+        """Query details of a DIS stream
+
+        :param stream_name: Name of the stream to which the checkpoint belongs.
+        :param app_name: Name of the app associated with the checkpoint.
+        :param partition_id: Identifier of the stream partition.
+        :param checkpoint_type: Type of the checkpoint (Default: LAST_READ).
+        :returns: One :class:`~otcextensions.sdk.dis.v2.checkpoint.Checkpoint`
+        :raises: :class:`~openstack.exceptions.ResourceNotFound`
+            when no resource can be found.
+        """
+        params = {
+            'stream_name': stream_name,
+            'app_name': app_name,
+            'partition_id': partition_id,
+            'checkpoint_type': checkpoint_type
+        }
+        # return self._get(
+        #    _checkpoint.Checkpoint, requires_id=False, params=params)
+        obj = self._get_resource(_checkpoint.Checkpoint, None)
+        return obj.get_checkpoint(self, **params)
+
+    def delete_checkpoint(self, stream_name, app_name,
+                          partition_id=None, checkpoint_type='LAST_READ'):
+        """Delete a CheckPoint Checkpoint
+
+        :param stream_name: Name of the stream to which the checkpoint belongs.
+        :param app_name: Name of the app associated with the checkpoint.
+        :param partition_id: Identifier of the stream partition (Default: None)
+        :param checkpoint_type: Type of the checkpoint (Default: LAST_READ).
+
+        :returns: ``None``
+        """
+        params = {
+            'stream_name': stream_name,
+            'app_name': app_name,
+            'checkpoint_type': checkpoint_type
+        }
+        if partition_id:
+            params.update(partition_id=partition_id)
+
+        # return self._delete(
+        #    _checkpoint.Checkpoint, requires_id=False, params=params)
+
+        obj = self._get_resource(_checkpoint.Checkpoint, None)
+        return obj.delete_checkpoint(self, **params)
+
+    # ======== Data Management ========
+    def upload_data(self, **attrs):
+        return self._create(_data.Data, **attrs)
```

Comment:
could you add some doc strings here and below?

Reply:
Done